### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by himadri2110**
- Login
- Signup
- Logout

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -18,6 +26,30 @@ const Navbar = () => {
...
+            to="/logout"
+            className="icon logout"
+            title="Logout"
+            onClick={() => {
+              localStorage.removeItem("VL_token");
+              localStorage.setItem("VL_isAuth", false);
+              dispatchAuth({
+                type: SET_AUTH,
+                payload: { isAuth: false },
+              });
```
Comment (msdeshmukh009) [inline src/components/Navbar/Navbar.jsx]: Maybe extract this function.

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+              </button>
+            ) : (
+              <button
+                className="btn btn-primary btn-disabled"
+                type="submit"
+                disabled
+              >
+                Create New Account
+              </button>
+            )}
```
Comment (msdeshmukh009) [inline src/pages/Authentication/SignUp/SignUp.jsx]: If I'm not wrong you can also pass an expression to the disabled attribute of the button which will take care of disabling the button. 
```jsx
 <button className=`btn btn-primary ${!signup.pwdMatch ? "btn-disabled":""}` 
              type="submit" 
             disabled={!signup.pwdMatch}>
                Create New Account
  </button>
```

**Review by msdeshmukh009 — APPROVED** [review]
The rest is looking fine. Nice work!!

**Review by percius47 — APPROVED** [review]
the main explore container falls out in mobile view, maybe that could be managed better. Apart from  that, auth look good to go.

File: src/components/Navbar/Navbar.jsx (hunk 1 above)
Comment (himadri2110) [reply src/components/Navbar/Navbar.jsx]: Will do that. Thanks.

**Comment by himadri2110** [comment]
> the main explore container falls out in mobile view, maybe that could be managed better. Apart from that, auth look good to go.

Haven't handled responsiveness as of now, will do it later. Thank you.

File: src/pages/Authentication/SignUp/SignUp.jsx (hunk 2 above)
Comment (himadri2110) [reply src/pages/Authentication/SignUp/SignUp.jsx]: Done, thanks for the suggestion.

**Pull request merged**